docs/developer/test_suite.md

```diff
@@ -0,0 +1,465 @@
+
+### Adding or changing tests
+
+Code sustainability requires that when developing any new functionality or extending existing capabilities, the test suite is also enhanced to test any new outputs.
+
+The changes required are
+
+  * (Optionally) a new test subdirectory. This will be automatically scanned by `testcode` and included if a subdirectory of one of the existing major categories. If the new output is an enhancement or option to an existing functionality, it may be appropriate to simply add the new test to an existing second-level subdirectory.
+  * When adding is a major new capability, it may be appropriate to add a new top-level category and subdirectory tree. In that case the [`Makefile`](#adding-to-make) and [`CMakeLists.txt`](#adding-to-cmake) will need to be extended to add the new `check`, `compare`, `diff` and `recheck` targets.
+  * CASTEP input files: `.cell` and `.param`
+  * If any external pseudopotential files are required, these should be placed in the `Test/Pseudopotentials` subdirectory and a symbolic link made from the test subdirectory. Both the fle and the link should be added to the `git` repository. 
+  * to modify [`bin/run_castep_test.py`](#test-outputs-and-benchmarks) if you need to test the contents of a new output file.
+  * if testing a new quantity, either in the `.castep` file or any other output file, you must extend the [`bin/extract_results.pl`](#extraction-of-properties) script to scan for and report the new value to be tested.
+  * A new Stanza in [`jobconfig`](#jobconfig-file) will be required to specify tolerances for the comparison if the defaults in [`userconfig`](#userconfig-file) are not suitable.
+  * a new benchmark file containing CASTEP output from a known good
+run. Unfortunately creating benchmark files still requires some manual
+intervention.  Regrettably `testcode`s option to create a new
+benchmark is unsuitable as it is too inflexible to create a single
+test without requiring updates to the entire test suite.  The easiest
+way to do this is
+
+    a. Run testcode, either from Make or by hand.  The new test will
+fail because no benchmark is present for comparison.
+
+    b. Copy or rename the just created test file
+"test.out.<date>?inp=<seedname>.param    to
+"benchmark.out.castep-<old-castep-version>castep-python-1.0.python-<old-python-version>.inp=<seedname>.param".
+where "<old-castep-version>" and "<old-python-version>" should be chosen to match the existing "benchmark.out" files from other tests.
```

Comment:
```suggestion
"test.out.<date>?inp=<seedname>.param"  to
"benchmark.out.castep-<old-castep-version>castep-python-1.0.python-<old-python-version>.inp=<seedname>.param".
where "<old-castep-version>" and "<old-python-version>" should be chosen to match the existing "benchmark.out" files from other tests.
```
Double check this renders since HTML style `<>` can be interpreted by Markdown. You *may* need to escape them.